CONTRIBUTING.md

```diff
@@ -0,0 +1,401 @@
+How to Contribute to Git for Windows
+====================================
+
+Git was originally designed for Unix systems and still today, all the build tools for the Git
+codebase assume you have standard Unix tools available in your path. If you have an open-source
+mindset and want to start contributing to Git, but primarily use a Windows machine, then you may
+have trouble getting started. This guide is for you.
+
+Get the Source
+--------------
+
+Clone the [GitForWindows repository on GitHub](https://github.com/git-for-windows/git).
+It is helpful create your own fork for storing your development branches.
+
+Windows uses CRLF line endings, while Unix systems just use LF. See
+[this GitHub article on working with line endings](https://help.github.com/articles/dealing-with-line-endings/#refreshing-a-repository-after-changing-line-endings)
+if you have trouble.
+
+Build the Source
+----------------
+
+First, download and install the latest [Git for Windows SDK (64-bit)](https://github.com/git-for-windows/build-extra/releases/latest).
+When complete, you can run the Git SDK, which creates a new Git Bash terminal window with
+the additional development commands, such as `make`.
+
+    The SDK uses a different credential manager, so you may still want to use Visual Studio
+    or normal Git Bash for interacting with your remotes.  Alternatively, use SSH rather
+    than HTTPS and avoid credential manager problems.
+
+You should now be ready to type `make` from the root of your `git` source directory.
+Here are some helpful variations:
+
+* `make -j[N] DEVELOPER=1`: Compile new sources using up to N concurrent processes.

```

Comment:
I propose to use `-j$(nproc)` in these `make` invocations. nproc(1) exists under both Linux and GfW and returns the number of online processors.

Reply:
That's a neat trick! I wonder if that is too painful to write every time, but I will definitely mention it.

Reply:
Maybe we should come up with a script that installs Git aliases for all these? But that might make things confusing for readers who jump right into the middle of the guide...

Reply:
What about `alias make='make -j$(nproc)'`? Pretty transparent IMO. Or integrate all these into the `sdk` function from #1357?

Reply:
I'd rather keep this document a bit independent of the Git for Windows SDK, as it might also help *non-Windows* developers get started with contributing to Git.